internal/plans/planfile/tfplan.go

```diff
@@ -823,6 +821,15 @@ func changeToTfplan(change *plans.ChangeSrc) (*planproto.Change, error) {
 	case plans.CreateThenDelete:
 		ret.Action = planproto.Action_CREATE_THEN_DELETE
 		ret.Values = []*planproto.DynamicValue{before, after}
+	case plans.Open:
+		ret.Action = planproto.Action_OPEN
+		// We need to write ephemeral resources to the plan file to be able to build
+		// the apply graph on `tofu apply <planfile>`.
+		// The DiffTransformer needs the changes from the plan to be able to generate
+		// executable resource instance graph nodes so we are adding the ephemeral resources too.
+		// Even though we are writing these, the actual values of the ephemeral *must not*
+		// be written to the plan so set nothing here.
```

Comment:
Perhaps a comment linking this to changeFromTfplan?

Reply:
Not sure that I get your point. This comment is already in `changeFromTfplan`. Do you mean to add a comment in DiffTransformer about `changeFromTfplan`? 🤔